components/collector/src/source_collectors/azure_devops/issues.py

```diff
@@ -52,9 +79,20 @@ async def _parse_value(self, responses: SourceResponses) -> Value:
 
         We can't just count the entities because due to pagination the response may not contain all work items.
         """
-        return str(len((await responses[0].json())["workItems"]))
+        return str(len(self._issue_ids_to_fetch))
+
+    def _include_issue(self, issue: dict) -> bool:  # pylint: disable=unused-argument # skipcq: PYL-R0201
+        """Return whether this issue should be counted."""
+        return True
 
-    @staticmethod
-    async def _work_items(responses: SourceResponses):
+    async def _work_items(self, responses: SourceResponses) -> list[dict]:
         """Return the work items, if any."""
-        return (await responses[1].json())["value"] if len(responses) > 1 else []
+        if len(responses) <= 1:  # the first call is to workItems, which only returns ids
+            return []
+        all_work_items = []
+        for response in responses[1:]:
+            response_json = (await response.json())
+            if not response_json:
+                continue
+            all_work_items.extend(response_json.get("value"))
```

Comment:
Why a guard clause if the work is only one line? How about this? Two fewer lines:
```suggestion
            if response_json := (await response.json()):
                all_work_items.extend(response_json.get("value"))
```